components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/TypeEncoders.scala

```diff
@@ -129,8 +137,13 @@ class TypingResultDecoder(loadClass: String => Class[_]) {
 
   private def typedObjectWithValue(obj: HCursor): Decoder.Result[TypingResult] = for {
     valueClass <- typedClass(obj)
-    value      <- SimpleObjectEncoder.decode(valueClass, obj.downField("value"))
-  } yield TypedObjectWithValue(valueClass, value)
+    value = SimpleObjectEncoder.decodeValue(valueClass, obj.downField("value"))
+  } yield value match {
+    case Left(e) =>
+      logger.warn(s"Failed value decoding: $e")
+      valueClass
```

Comment:
We we need this fallback? Generally, it is usually better to fail fast. Thanks to that it is easier to detect the root problem and fix the bug.

Reply:
I thought that maybe if we encounter a value that we can't encode/decode, but we can encode/decode the class, then returning the class alone (and logging the failure) is better/safer than throwing

If you have the opposite opinion, I'll change it